core/server/helpers/navigation.js

```diff
@@ -0,0 +1,58 @@
+// ### Navigation Helper
+// `{{navigation}}`
+// Outputs navigation menu of static urls
+
+var _               = require('lodash'),
+    hbs             = require('express-hbs'),
+    errors          = require('../errors'),
+    template        = require('./template'),
+    navigation;
+
+navigation = function (options) {
+    /*jshint unused:false*/
+    var nav,
+        context,
+        currentUrl = this.relativeUrl;
+
+    if (!_.isObject(this.nav) || _.isFunction(this.nav)) {
+        return errors.logAndThrowError('navigation data is not an object or is a function');
+    }
+
+    if (this.nav.filter(function (e) {
+            return (_.isUndefined(e.label) || _.isUndefined(e.url));
+        }).length > 0) {
+            return errors.logAndThrowError('All values must be defined for label, url and current');
+        }
+
+    // check for non-null string values
+    if (this.nav.filter(function (e) {
+        return ((!_.isNull(e.label) && !_.isString(e.label)) ||
+            (!_.isNull(e.url) && !_.isString(e.url)));
+    }).length > 0) {
+        return errors.logAndThrowError('Invalid value, Url and Label must be strings');
+    }
+
+    function _slugify(label) {
+        return label.toLowerCase().replace(/[^\w ]+/g, '').replace(/ +/g, '-');
+    }
+
+    // {{navigation}} should noop if no data passed in
+    if (this.nav.length === 0) {
+        return {string: ''};
```

Comment:
@ErisDS small question: this return value is the typical return value from `templates.execute` but it feels a bit weird to return this here. happy to leave as is, since it no longer outputs an empty list for empty nav, but maybe there's a more accepted way of doing this?